File: Objektorientierung/VerketteteListenRuby/verketteteListe.rb
require "./verkettetesElement.rb"

class VerketteteListe
    def initialize()
        @startElement = nil
        @letztesElement = nil
    end

    def elementHinzufuegen(w)
        if @startElement == nil then
            @startElement = VerkettetesElement.new(w)
            @letztesElement = @startElement
        else
            neuesElement = VerkettetesElement.new(w)
            neuesElement.vorgaenger=@letztesElement
            @letztesElement.nachfolger=neuesElement
            @letztesElement = neuesElement
        end
    end

    def alleElementeAusgeben()
        iterator = @startElement
        while(iterator != nil) do
             puts(iterator.wert())
             iterator = iterator.nachfolger()
        end
     end
 

    def letztesElementLoeschen()

    end

    def laengeDerListe()

    end

    def istEintragInListe()

    end

end

File: Objektorientierung/VerketteteListenPython/verketteteListe.py
import verkettetesElement

class VerketteteListe:
    def __init__(self):
        self.startElement = None
        self.letztesElement = None

    def elementHinzufuegen(self,w):
        if self.startElement == None:
            self.startElement = verkettetesElement.VerkettetesElement(w)
            self.letztesElement = self.startElement
        else:
            neuesElement = verkettetesElement.VerkettetesElement(w)
            neuesElement.setVorgaenger(self.letztesElement)
            self.letztesElement.setNachfolger(neuesElement)
            self.letztesElement = neuesElement

    def alleElementeAusgeben(self):
        iterator = self.startElement
        while iterator != None:
             print(iterator.getWert())
             iterator = iterator.getNachfolger()
 

    #def letztesElementLoeschen(self):
        
    #def laengeDerListe(self):
    
    #def istEintragInListe(self):
